internal/impl/shell.go

```diff
@@ -87,28 +88,29 @@ func NewDevboxShell(nixpkgsCommitHash string, opts ...ShellOption) (*DevboxShell
 }
 
 // shellPath returns the path to a shell binary, or error if none found.
-func shellPath(nixpkgsCommitHash string) (path string, err error) {
+func shellPath(nixpkgsCommitHash string, pure bool) (path string, err error) {
 	defer func() {
 		if err != nil {
 			path = filepath.Clean(path)
 		}
 	}()
 
-	// First, check the SHELL environment variable.
-	path = os.Getenv(envir.Shell)
-	if path != "" {
-		debug.Log("Using SHELL env var for shell binary path: %s\n", path)
-		return path, nil
+	if !pure {
+		// First, check the SHELL environment variable.
+		path = os.Getenv(envir.Shell)
+		if path != "" {
+			debug.Log("Using SHELL env var for shell binary path: %s\n", path)
+			return path, nil
+		}
```

Comment:
I think here we should ignore it and always use Nix's bash shell. IMO the shell can have a pretty big impact on the environment.

I agree with the use-case of letting the user override the shell, but it might make more sense to do that through a new devbox.json field. That way it can be a Nix-installed shell that's guaranteed to be the same for everyone.